en/tutorials-and-examples/cms/authentication-plugin.rst

```diff
@@ -0,0 +1,288 @@
+            }
+        }
+    }
+
+Now, point your browser to **http://localhost:8765/users** to see a list of users.
+Remember you'll need to have your local server running, start a standalone PHP
+server using ``bin/cake server``
+
+You can edit the default user that was created during
+:doc:`Installation <installation>`. If you change that user's password,
+you should see a hashed password instead of the original value on the list or
+view pages. CakePHP hashes passwords with `bcrypt
+<http://codahale.com/how-to-safely-store-a-password/>`_ by default. We recommend
+bcrypt for all new applications to keep your security standards high.
+
+.. note::
+
+    Create a hashed password for at least one of the user accounts now! 
+    It will be needed in the next steps.
+    After updating the password, you'll see a long string stored in the password column.
+    Note bcrypt will generate a different hash even for the same password saved twice.
+    
+
+Installing Authentication Plugin
+================================
+
+Use composer to install the Authentication Plugin
+
+    composer require cakephp/authentication:^2.0
+
+Adding Login
+============
+
+Now it's time to configure the Authentication Plugin.
+The Plugin will handle the authentication process using 3 different classes:
+
+* ``Application`` will use the Authentication Middleware and provide an
+AuthenticationService, holding all the configuration we want to define how are
+we going to check the credentials, and where to find them.
+* ``AuthenticationService`` will be a utility class to allow you configure the
+authentication process.
+* ``AuthenticationMiddleware`` will be executed as part of the middleware queue,
+this is before your Controllers are processed by the framework, and will pick the
+credentials and process them to check if the user is authenticated.
+
+If you remember, we used :doc:`AuthComponent </controllers/components/authentication>`
+before to handle all these steps. Now the logic is divided into specific classes and
+the authentication process happens before your controller layer, checking if the user
+is authenticated (based in the configuration you provided) and injecting the user and
+the authentication results in the request for further reference.
+
+In you Application class add the following code
+
+    // src/Application.php
+    public function middleware(MiddlewareQueue $middlewareQueue): MiddlewareQueue
+    {
+        $middlewareQueue
+            // ... other middleware added before
+            ->add(new RoutingMiddleware($this))
+            // add Authentication after RoutingMiddleware
+            ->add(new \Authentication\Middleware\AuthenticationMiddleware($this->configAuth(), [
+                'unauthenticatedRedirect' => '/users/login',
+                'queryParam' => 'redirect',
+            ]));
+
+        return $middlewareQueue;
+    }
+
+    protected function configAuth(): \Authentication\AuthenticationService
```

Comment:
Should we recommend using the provider interface? It is more boilerplate, so perhaps this approach is better.